ui-lib/src/components/ProgressBar/index.tsx

```diff
@@ -0,0 +1,157 @@
+   */
+  steps: string[];
+  /**
+   * Index of the active step, starting at zero and defaulting to 0 if omitted.
+   */
+  activeIndex?: number;
+}
+/**
+ * The state of a progress bar step.
+ */
+type StepState = "pending" | "current" | "completed";
+export type StepProps = {
+  /**
+   * State of the step.
+   */
+  state: StepState;
+  /**
+   * The label to display.
+   */
+  label: string;
+  /**
+   * If true, hides the left "separator" bar that bridges the gap between the wider separator and the item.
+   */
+  hideLeftSeparator: boolean;
+  /**
+   * The next step state, or undefined if the current step is the last one.
+   */
+  nextState?: StepState;
+};
+
+export const Item = {
+  Container: styled.div.attrs(_ => ({
+    mx: 2,
+  }))<ColorProps & BorderProps & SpaceProps>`
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: 24px;
+    height: 24px;
+    ${color}
+    ${border}
+    ${space}
+  `,
+  Spacer: styled.div<SpaceProps>`
+    display: flex;
+    align-self: stretch;
+    ${space}
+  `,
+  Current: styled.div.attrs(_ => ({
+    backgroundColor: "palette.v2.primary.dark",
+  }))<ColorProps>`
+    width: 6px;
+    height: 6px;
+    border-radius: 9999px;
+    ${color}
+  `,
+  Pending: styled.div.attrs(_ => ({
+    backgroundColor: "palette.v2.text.secondary",
+  }))<ColorProps>`
+    width: 4px;
+    height: 4px;
+    border-radius: 9999px;
+    ${color}
+  `,
+  Completed: (): JSX.Element => <CheckAlone size={16} />,
+};
+
+const StepText = styled(Text)<{ inactive?: boolean }>`
+  color: ${p =>
+    p.inactive ? p.theme.colors.palette.v2.text.secondary : p.theme.colors.palette.v2.text.default};
+`;
+
+const BaseSeparator = styled.div<{ inactive?: boolean }>`
+  border: 1px solid;
+  height: 0px;
+  border-color: ${p =>
+    p.inactive ? p.theme.colors.palette.v2.grey.border : p.theme.colors.palette.v2.text.default};
+`;
+const Separator = {
+  Step: styled(BaseSeparator)`
+    flex: 1;
+    position: relative;
+    top: 12px;
+  `,
+  Item: styled(BaseSeparator)`
+    flex: 1;
+    top: 12px;
+    position: relative;
+  `,
+};
+
+export const Step = memo(function Step({
+  state,
+  label,
+  hideLeftSeparator,
+  nextState,
+}: StepProps): JSX.Element {
+  const inactive = state === "pending";
+  const nextInactive = nextState === "pending";
+  return (
+    <Flex flexDirection="column" alignItems="center">
+      <Item.Spacer mb={2}>
+        {(!hideLeftSeparator && <Separator.Item inactive={inactive} />) || <Flex flex="1" />}
+        {state === "pending" ? (
+          <Item.Container>
+            <Item.Pending />
+          </Item.Container>
+        ) : state === "current" ? (
+          <Item.Container backgroundColor="palette.v2.primary.backgroundLight" borderRadius="8px">
+            <Item.Current />
+          </Item.Container>
+        ) : (
+          <Item.Container color="palette.v2.text.default">
+            <Item.Completed />
+          </Item.Container>
+        )}
```

Comment:
Maybe it's only me but I'm not a big fan of using nested ternary from the JSX tbh.
I generally prefer using a "switch" component that returns different JSX based on the state value.

Just my two cents, we don't have rules about that so it's not a change request.

Reply:
> I generally prefer using a "switch" component that returns different JSX based on the state value.

No strong opinion on this, I use both depending on the context. In this case I find it readable enough but I'll happily follow the guidelines if we define some project-wide rules.

Reply:
We use ternary galore throughout all the projects, I think this is a decision that should be made by live-core and not us, so feel free to raise the point to the team and in the meantime, do whatever makes you happy